lib/rom/http/dataset.rb

```diff
@@ -0,0 +1,103 @@
+module ROM
+  module HTTP
+    class Dataset
+      include Enumerable
+
+      attr_reader :config, :options
+
+      def initialize(config, options = {})
+        @config = config
+        @options = {
+          request_method: :get,
+          path: '',
+          params: {}
+        }.merge(options)
+      end
+
+      def url
+        config[:url]
+      end
+
+      def name
+        config[:name]
+      end
+
+      def path
+        options[:path].to_s
+      end
+
+      def request_method
+        options[:request_method]
+      end
+
+      def params
+        options[:params]
+      end
+
+      def with_options(opts)
+        self.class.new(config, options.merge(opts))
+      end
+
+      def with_path(path)
+        with_options(path: path)
+      end
+
+      def prepend_path(path)
+        with_options(path: File.join(path, options[:path]))
```

Comment:
Think in this case `File.join` does a better job than `Addressable` or `URI`, however, it might be better to have some kind of wrapper to interact with rather than the current implementation.